src/OceanSeaIceModels/CrossRealmFluxes/similarity_theory_turbulent_fluxes.jl

```diff
@@ -360,45 +374,88 @@ end
 @inline velocity_differences(𝒰₁, 𝒰₀, ::RelativeVelocity) = @inbounds 𝒰₁.u[1] - 𝒰₀.u[1], 𝒰₁.u[2] - 𝒰₀.u[2]
 @inline velocity_differences(𝒰₁, 𝒰₀, ::WindVelocity)     = @inbounds 𝒰₁.u[1], 𝒰₁.u[2] 
 
-@inline function state_differences(ℂ, 𝒰₁, 𝒰₀, g, bulk_velocity)
+@inline function state_differences(ℂ, 𝒰₁, 𝒰₀, θ₀, S₀, Σ★, g, ρₒ, cpₒ, 
+                                   water_mole_fraction,
+                                   water_vapor_saturation,
+                                   surface_temperature_type, 
+                                   prescribed_heat_fluxes,
+                                   radiative_properties,
+                                   bulk_velocity)
     z₁ = 𝒰₁.z
     z₀ = 𝒰₀.z
     Δh = z₁ - z₀
     Δu, Δv = velocity_differences(𝒰₁, 𝒰₀, bulk_velocity)
-
+    
     # Thermodynamic state
     𝒬₁ = 𝒰₁.ts
     𝒬₀ = 𝒰₀.ts
 
-    θ₁ = AtmosphericThermodynamics.air_temperature(ℂ, 𝒬₁)
-    θ₀ = AtmosphericThermodynamics.air_temperature(ℂ, 𝒬₀)
+    ρₐ = AtmosphericThermodynamics.air_density(ℂ, 𝒬₁)
     cₚ = AtmosphericThermodynamics.cp_m(ℂ, 𝒬₁) # moist heat capacity
+    ℰv = AtmosphericThermodynamics.latent_heat_vapor(ℂ, 𝒬₁)
+
+    θ₀ = compute_surface_temperature(surface_temperature_type, θ₀, ℂ, 𝒬₀, ρₐ, cₚ, ℰv, Σ★, ρₒ, cpₒ, g,
+                                     prescribed_heat_fluxes, 
+                                     radiative_properties)
+
+    θ₁ = AtmosphericThermodynamics.air_temperature(ℂ, 𝒬₁)
 
     # Temperature difference including the ``lapse rate'' `α = g / cₚ`
     Δθ = θ₁ - θ₀ + g / cₚ * Δh
 
     q₁ = AtmosphericThermodynamics.vapor_specific_humidity(ℂ, 𝒬₁)
-    q₀ = AtmosphericThermodynamics.vapor_specific_humidity(ℂ, 𝒬₀)
+
+    # Recomputing the saturation specific humidity at the surface based on the new temperature
+    q₀ = seawater_saturation_specific_humidity(ℂ, θ₀, S₀, 𝒬₁,
+                                               water_mole_fraction,
+                                               water_vapor_saturation,
+                                               AtmosphericThermodynamics.Liquid())
+    
+    𝒬ₛ = AtmosphericThermodynamics.PhaseEquil_pTq(ℂ, 𝒬₀.p, θ₀, q₀)
+    q₀ = AtmosphericThermodynamics.vapor_specific_humidity(ℂ, 𝒬ₛ)
```

Comment:
You don't want to call this again, because the "seawater" part is important --- we account for the h2O mole fraction there. Also, we may be using a different saturation parameterization than `AtmosphericThermodynamics` (this is to comply with OMIP protocol)

Reply:
hmm ok, if I remove that line the results are different (regression tests do not pass). Isn't ok that the seawater part called above?

Reply:
weird, not sure I understand what is going on then

Reply:
I mean, were the regression tests also incorrect?

Reply:
Or maybe I am confused. Does `vapor_specific_humidity(ℂ, 𝒬ₛ)` compute a new specific humidity or merely extract it?

Reply:
I think I'm confused sorry, I guess `vapor_specific_humidity(ℂ, 𝒬ₛ)` just extracts it. It says nothing about saturation there.

Reply:
But I am weirded out that it changes the result. What is the difference between the two q0?

Reply:
then probably yes, because in the main code we do:
https://github.com/CliMA/ClimaOcean.jl/blob/0fff4fa3564d50a5f44ba7fe45b6e647242387b8/src/OceanSeaIceModels/CrossRealmFluxes/similarity_theory_turbulent_fluxes.jl#L380-L382 
where `𝒬₀` holds the previously calculated `q₀`

Reply:
Ok, on `main` what we do is compute q0, and then we assemble it into a thermodynamic / dynamic state. Then we pass the whole state to the solver. Within the solver, I guess it uses `vapor_specific_humidity` to extract q0 --- or at least, that's what I thought. This result suggests though that somehow q0 is changed?

Reply:
I think it is changed in the `vapor_specific_humidity` function. It is kind of tough to follow the thread, but it looks like the `vapor_specific_humidity` builds a `PhasePartition` type to calculate the humidity vapor fraction associated with the thermodynamic state